### PR TITLE
Shopify adapters test

### PR DESCRIPTION
**Pull request by starikovskii**

### CODE REVIEW

**Review by DenysZP — APPROVED** [review]

File: library/src/test/java/com/shopapp/shopify/api/adapter/CategoryAdapterTest.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import com.shopapp.shopify.StorefrontMockInstantiator
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.junit.MockitoJUnitRunner
+
+@RunWith(MockitoJUnitRunner.Silent::class)
+class CategoryAdapterTest {
```
Comment (plzen) [inline library/src/test/java/com/shopapp/shopify/api/adapter/CategoryAdapterTest.kt:15]: Add test for second condition in `collection.descriptionHtml ?: DEFAULT_STRING`

File: library/src/test/java/com/shopapp/shopify/api/adapter/CheckoutAdapterTest.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import com.shopapp.shopify.JodaTimeAndroidRule
+import com.shopapp.shopify.StorefrontMockInstantiator
+import org.junit.Assert.*
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.junit.MockitoJUnitRunner
+
+@RunWith(MockitoJUnitRunner::class)
+class CheckoutAdapterTest {
```
Comment (plzen) [inline library/src/test/java/com/shopapp/shopify/api/adapter/CheckoutAdapterTest.kt:13]: Add tests for 

```
adaptee.shippingAddress?.let { AddressAdapter.adapt(it) },
adaptee.shippingLine?.let { ShippingRateAdapter.adapt(it) }
```

File: library/src/test/java/com/shopapp/shopify/api/adapter/CountryListAdapterTest.kt [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import com.shopapp.shopify.StorefrontMockInstantiator
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.junit.MockitoJUnitRunner
+
+@RunWith(MockitoJUnitRunner::class)
+class CountryListAdapterTest {
```
Comment (plzen) [inline library/src/test/java/com/shopapp/shopify/api/adapter/CountryListAdapterTest.kt:13]: Add test for second condition in `countries?.map { CountryAdapter.adapt(it) } ?: listOf()`

File: library/src/test/java/com/shopapp/shopify/api/adapter/ErrorAdapterTest.kt [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import com.shopify.buy3.GraphNetworkError
+import com.shopify.buy3.Storefront
+import org.junit.Assert.*
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.junit.MockitoJUnitRunner
+
+@RunWith(MockitoJUnitRunner::class)
+class ErrorAdapterTest {
```
Comment (plzen) [inline library/src/test/java/com/shopapp/shopify/api/adapter/ErrorAdapterTest.kt:16]: Add tests for

```
adaptee.firstOrNull()?.let { Error.NonCritical(it.message()) }
---
return adaptee?.let {
    it.message?.let { Error.NonCritical(it) }
}
---
adaptee.firstOrNull()?.let { Error.NonCritical(it.message) }
```

File: library/src/test/java/com/shopapp/shopify/api/adapter/PolicyAdapterTest.kt [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+import com.shopapp.shopify.JodaTimeAndroidRule
+import com.shopapp.shopify.StorefrontMockInstantiator
+import org.junit.Assert.assertEquals
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.junit.MockitoJUnitRunner
+
+@RunWith(MockitoJUnitRunner::class)
+class PolicyAdapterTest {
```
Comment (plzen) [inline library/src/test/java/com/shopapp/shopify/api/adapter/PolicyAdapterTest.kt:13]: Add test for second condition in `return if (adaptee != null) {`

File: library/src/test/java/com/shopapp/shopify/api/adapter/ProductAdapterTest.kt [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.junit.MockitoJUnitRunner
+import java.math.BigDecimal
+
+@RunWith(MockitoJUnitRunner.Silent::class)
+class ProductAdapterTest {
```
Comment (plzen) [inline library/src/test/java/com/shopapp/shopify/api/adapter/ProductAdapterTest.kt:18]: Add tests for

```
productAdaptee.variants.edges.mapNotNull { ProductVariantAdapter.adapt(it.node) }
---
return if (variantsList.size > 0) {
    val mappedList = variantsList.mapNotNull { it.node.price }
    Pair(mappedList.min() ?: DEFAULT_PRICE, mappedList.max() ?: DEFAULT_PRICE)
---
productAdaptee.images.edges.mapNotNull  { ImageAdapter.adapt(it.node)!! }
---
return if (options != null) {
```

File: library/src/test/java/com/shopapp/shopify/api/adapter/StateListAdapterTest.kt [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+import com.shopapp.shopify.StorefrontMockInstantiator
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.junit.MockitoJUnitRunner
+
+@RunWith(MockitoJUnitRunner::class)
+class StateListAdapterTest {
```
Comment (plzen) [inline library/src/test/java/com/shopapp/shopify/api/adapter/StateListAdapterTest.kt:16]: Add test for second condition in `states?.map { StateAdapter.adapt(it) } ?: listOf()`

**Review by plzen — CHANGES_REQUESTED** [review]

**Review by plzen — CHANGES_REQUESTED** [review]
Also add tests for `ProductExtension.kt`

**Review by plzen — APPROVED** [review]
